### PR TITLE
[22.01] Drop OpenID post authentication protocols

### PR DESCRIPTION
**Pull request by mvdbeek**
I think we only used this for genomespace, which is gone for a while now.
Fixes https://github.com/galaxyproject/galaxy/issues/13280 by removing the controller in question (which looks like it wasn't wired up with a route ?!)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: doc/source/admin/galaxy_options.rst [hunk 1]
```diff
@@ -4823,13 +4801,16 @@
...
 
 :Description:
-    Vault configuration
-    The value of this option will be resolved with respect to
-    <config_dir>.
+    Vault config file. The value of this option will be resolved with
+    respect to <config_dir>.
 :Default: ``vault_conf.yml``
 :Type: str
+
```
Comment (mvdbeek) [inline doc/source/admin/galaxy_options.rst:4815]: If someone knows where those 3 newlines are coming from I'd be so happy ... 🙏

Comment (nsoranzo) [reply doc/source/admin/galaxy_options.rst:4815]: It looks like they were actually removed (manually?) in https://github.com/galaxyproject/galaxy/pull/12940/files#diff-63e15fe9fdbbfc026e332aff2ecd8f44c56a7af1ba315df61e86bce8fd4905e8 , so nothing to worry about?

Comment (dannon) [reply doc/source/admin/galaxy_options.rst:4815]: I had these show up in another config rebuild I did recently, too -- definitely nothing to worry about.

Comment (mvdbeek) [reply doc/source/admin/galaxy_options.rst:4815]: Alright, then I'll drop them manually

Comment (dannon) [reply doc/source/admin/galaxy_options.rst:4815]: Do they mess anything up?  I'd just leave them as whatever gets automatically built w/ make config-rebuild.

File: lib/galaxy/config/schemas/config_schema.yml [hunk 2]
```diff
@@ -3518,4 +3503,4 @@ mapping:
         path_resolves_to: config_dir
         required: false
         desc: |
-          Vault config file.
+          Vault config file.
```
Comment (nsoranzo) [inline lib/galaxy/config/schemas/config_schema.yml]: I'd keep the newline here.

File: lib/galaxy/managers/configuration.py [hunk 3]
```diff
@@ -158,7 +157,6 @@ def _config_is_truthy(item, key, **context):
             'allow_user_creation': _defaults_to(False),  # schema default is True
             'use_remote_user': _defaults_to(None),  # schema default is False; or config.single_user
             'single_user': _config_is_truthy,
-            'enable_oidc': _use_config,
```
Comment (nsoranzo) [inline lib/galaxy/managers/configuration.py]: `enable_oidc` seems to still be used in a number of places, removed by mistake?

Comment (mvdbeek) [reply lib/galaxy/managers/configuration.py]: ah, yes!

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Not sure what's the issue with the startup test failures, but they seem reproducible.

File: lib/galaxy/config/sample/galaxy.yml.sample [hunk 4]
```diff
@@ -2353,7 +2345,8 @@ galaxy:
   # is relative to galaxy/static
   #welcome_directory: plugins/welcome_page/new_user/static/topics/
 
-  # Vault config file
+  # Vault config file.
   # The value of this option will be resolved with respect to
   # <config_dir>.
-  vault_config_file: vault_conf.yml
+  #vault_config_file: vault_conf.yml
```
Comment (mvdbeek) [inline lib/galaxy/config/sample/galaxy.yml.sample:2351]: @nsoranzo that's the issue, we now don't have
```yaml
galaxy:
  vault_config_file: vault_conf.yml
```
but
```yaml
galaxy:
```
which is None, and gravity checks for None in https://github.com/galaxyproject/gravity/blob/main/gravity/config_manager.py#L352

We can probably just make sure there's a galaxy key and not make any assumptions about the value.

Comment (nsoranzo) [reply lib/galaxy/config/sample/galaxy.yml.sample:2351]: Exactly, came to the same conclusion! I can open a PR for Gravity, if you haven't one ready yet.

Comment (mvdbeek) [reply lib/galaxy/config/sample/galaxy.yml.sample:2351]: Nope, thank you!

Comment (nsoranzo) [reply lib/galaxy/config/sample/galaxy.yml.sample:2351]: https://github.com/galaxyproject/gravity/pull/25

**Pull request merged**